src/main/groovy/com/github/prokod/gradle/crossbuild/utils/DependencyInsights.groovy

```diff
@@ -443,10 +444,9 @@ class DependencyInsights {
             def currentProjectTypeDependenciesDependencies = currentProjectTypDepsForDefault.collectMany { prjDep ->
                 extractCrossBuildProjectTypeDependencyDependencies(prjDep, configurationNames)
             }
-            accum.addAll(extractCrossBuildProjectTypeDependenciesRecursively(modules,
-                    currentProjectTypeDependenciesDependencies.toSet(), configurationNames))
+            extractCrossBuildProjectTypeDependenciesRecursively(modules,
```

Comment:
we pass the accum down anyway so no need to add them all again.

Reply:
You do the same as me but you expose it to the function signature (as a parameter). I did not need that option in code so I preferred to reduce amount of params (even though you defaulted it to empty :) )

Reply:
It isn't the same, because with my change you only check a sub project only once. Because you filter with the full known set of things you have seen and not with the set you just found.

Reply:
Are you referring to the extra filtering you do or to the fact I did `accum.addAll` ?

Reply:
I'm refering to the combination of the extra filter and passing the accum down the accum is used to know what it had already visited and what not.

so in the example case i put in issue: 
https://github.com/prokod/gradle-crossbuild-scala/issues/71

In the old version: if it starts resolving A it then picks up B and then it detects A again and visits it to resolve its dependencies, goes back to B etc. and keeps recursing. Because it didn't pass the Accum to keep track of what it had visited. 

in the new Version:  It passes the accum down so it starts at A, add A to the accum, go down to B and sees A again but with the accum it knows that it saw A so it will not resolve it again. So that resolves the recursion problem. 

An added advantage is that it's also more efficient since it only checks the dependencies once.